util/dvsim/SimCfg.py

```diff
@@ -590,7 +589,7 @@ def create_failure_message(test, line, context):
             self.results_summary = self.testplan.results_summary
 
             # Append coverage results if coverage was enabled.
-            if self.cov_report_deploy is not None:
+            if self.cov_report_deploy:
```

Comment:
That's interesting. Do we want to treat the empty string specially here? Naively, this doesn't look like it would be related to what the commit message says the patch is doing. (Similarly in `_publish_results`)

Reply:
Cool! I think I wasn't clear. What I meant is that I'd expect the check to be string vs. None (which is what the code was doing before) and I was wondering whether you had started needing to treat the empty string specially. I definitely prefer the explicit None check, thanks.